util/net/http_transport_mac.mm

```diff
@@ -283,13 +283,20 @@ static bool ExecuteProxyRequest(NSMutableURLRequest* request,
     NSNumber* proxy_port = @(std::stoi(port));
 
     NSDictionary* proxyDict = @{
-      (__bridge id)kCFNetworkProxiesHTTPEnable : @YES,
-      (__bridge id)kCFNetworkProxiesHTTPPort : proxy_port,
-      (__bridge id)kCFNetworkProxiesHTTPProxy : hostNS,
+      @"HTTPEnable" : @YES,
+      @"HTTPPort" : proxy_port,
+      @"HTTPProxy" : hostNS,
       @"HTTPSEnable" : @YES,
       @"HTTPSPort" : proxy_port,
       @"HTTPSProxy" : hostNS,
     };
```

Comment:
These would still be set if the scheme is `socks5`, right? They should also be conditionally set if the scheme is HTTP or HTTPS.

Reply:
You mean a condition for each of the three proxy schemes? I now only placed a condition on the socks5 one, since in the old version both HTTP and HTTPS would be in the dict.

Reply:
You are correct that the current initialization already covers the other case (http+https). Nonetheless, I'd prefer for this to be explicit and not only a fall-back in the socks5 case:

```objectivec
NSDictionary* proxyDict;

if ([schemeNS isEqualToString:@"http"] || [schemeNS isEqualToString:@"https"]) {
    proxyDict = @{
      @"HTTPEnable" : @YES,
      @"HTTPPort" : proxy_port,
      @"HTTPProxy" : hostNS,
      @"HTTPSEnable" : @YES,
      @"HTTPSPort" : proxy_port,
      @"HTTPSProxy" : hostNS
    };
} else if ([schemeNS isEqualToString:@"socks5"]) {
    proxyDict = @{
      @"SOCKSEnable" : @YES,
      @"SOCKSPort" : proxy_port,
      @"SOCKSProxy" : hostNS
    };
}
```

Make fun all you want. This code is so rarely touched that whoever gets to see it next will be happy about this being explicit.